doc: release-notes: native targets related updates

aescolar opened the pull request:
Add to the 3.6 release notes the native targets related updates.
And mention that we do not define the SOURCE macros globally anymore

aescolar committed:
doc: release-notes: Mention removal of global SOURCE definitions

Mention that we do not define these macros globally anymore

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

aescolar committed:
doc: release-notes: native targets related updates

Add to the 3.6 release notes the native targets related
updates.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

kartben commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -147,6 +167,11 @@ Build system and infrastructure
 
+* Fixed an issue where the macros ``_POSIX_C_SOURCE`` and ``_XOPEN_SOURCE`` would be defined
+  globally when building with Picolibc or for the native (``ARCH_POSIX``) targets.
+  After this change users may need to define them for their own applications or libraries if they
+  require them.
```
Is this worth an entry in the migration guide? I have a downstream app whose CI just broke so I guess it might happen to others, and I could definitely use some guidance there myself :) 

```
...
/Users/kartben/artificial-nose-zephyr-ws/zephyr/drivers/wifi/esp32/src/esp_wifi_drv.c: In function 'scan_done_handler':
/Users/kartben/artificial-nose-zephyr-ws/zephyr/drivers/wifi/esp32/src/esp_wifi_drv.c:204:40: error: implicit declaration of function 'strnlen'; did you mean 'strlen'? [-Werror=implicit-function-declaration]
  204 |                         int ssid_len = strnlen(ap_list_buffer[k].ssid, WIFI_SSID_MAX_LEN);
      |                                        ^~~~~~~
      |                                        strlen
cc1: all warnings being treated as errors
...
```

aescolar replied:
Oh, that's a bug (and it got in in a PR of mine :( ) fix posted here https://github.com/zephyrproject-rtos/zephyr/pull/68233 
Anyhow, it does make sense to mention this in the migration guide, so I will.

kartben commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -413,5 +449,14 @@ Tests and Samples
 
+* Bluetooth split stacks tests, where the BT host and controller are run in separate MCUs, are
+  now run in CI based on the :ref:`nrf5340_bsim<nrf5340bsim>` targets
+  Several other runtime AMP tests based on these targets have been added to CI. Including tests
+  of OpenAMP, the mbox and IPC drivers/subsystem and the logger multidomain functionality.
```
```suggestion
  Several other runtime AMP tests based on these targets have been added to CI, including tests
  of OpenAMP, the mbox and IPC drivers/subsystem, and the logger multidomain functionality.
```

kartben reviewed (changes requested):
This is great, thanks for this!
A minor grammar nit and a comment.

aescolar committed:
doc: releases: migration-guide: 3.6: Add note on SOURCE macros

Warn users about the SOURCE macros not being defined globally anymore
and them needed to set them if they need them.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

kartben commented on doc/releases/release-notes-3.6.rst:
```diff
@@ -413,5 +449,14 @@ Tests and Samples
   in a future release.
 
+* Bluetooth split stacks tests, where the BT host and controller are run in separate MCUs, are
+  now run in CI based on the :ref:`nrf5340_bsim<nrf5340bsim>` targets
```
```suggestion
  now run in CI based on the :ref:`nrf5340_bsim<nrf5340bsim>` targets.
```

kartben reviewed (changes requested):
Thanks for adding the migration guide, looks good!
